avro-fastserde/src/main/java/com/linkedin/avro/fastserde/Utils.java

```diff
@@ -26,6 +28,15 @@
   // Cache the mapping between Schema and the corresponding fingerprint
   private static final Map<Schema, Long> SCHEMA_IDS_CACHE = new ConcurrentHashMap<>();
 
+  // Schema#hashCode is not performance efficient in Avro 1.4. It has been improved
+  // by caching hashcode in the later version. Change to use WeakIdentityHashMap for
+  // Avro 1.4 helps to improve the cache lookup performance. However, it requires
+  // the client to resuse the schame instance.
```

Comment:
nit - comments about need for LRU no longer true?

Reply:
We still need this improvement in the future. It' not only related to this PR. The de/serializer cache will also lead to memory issue if user have created a great number of schemas, such as in projection scenario...